Jenkinsfile

```diff
@@ -271,10 +271,41 @@ builders = pipeline_builder.createBuilders { container ->
         container.sh """
           cd build
           pwd
-          ninja docs 2> ${test_output}
+          ninja docs 2>&1 > ${test_output}
         """
         container.copyFrom("build/${test_output}", '.')
+
+        String regexMissingDocsMemberConsumer = /.* warning: Member Consumer .* of class .* is not documented .*/
+        String regexInvalidArg = /.* warning: argument: .* is not found in the argument list of .*/
+        String regexMissingCompoundDocs = /.*warning: Compound .* is not documented./
+        boolean doxygenStepFailed = false
+        def failingCases = []
+        def doxygenResultContent = readFile test_output
+        def doxygenResultLines = doxygenResultContent.readLines()
+
+        for(line in doxygenResultLines) {
+            if(line ==~ regexMissingDocsMemberConsumer || line ==~ regexInvalidArg ||
+                line ==~ regexMissingCompoundDocs) {
+                failingCases.add(line)
+            }
+        }
+
+        int numFailedCases = failingCases.size()
+        if(numFailedCases) {
+            doxygenStepFailed = true
+            println "Doxygen failed to generate HTML documentation due to following warnings:"
+            for(line in failingCases) {
+                println line
+            }
+            println "Total number of Doxygen warnings that needs to be corrected is $numFailedCases."
```

Comment:
I think this line should be printed before the actual warning messages so as to make it easier for you to get an overview on how much work you have to do.